diff -rupN pki-core-9.0.3.orig/base/ca/shared/conf/CS.cfg.in pki-core-9.0.3/base/ca/shared/conf/CS.cfg.in
--- pki-core-9.0.3.orig/base/ca/shared/conf/CS.cfg.in	2014-06-19 21:01:06.000000000 -0700
+++ pki-core-9.0.3/base/ca/shared/conf/CS.cfg.in	2014-06-19 21:09:23.000000000 -0700
@@ -783,7 +783,7 @@ debug.append=true
 debug.enabled=true
 debug.filename=[PKI_INSTANCE_PATH]/logs/debug
 debug.hashkeytypes=
-debug.level=0
+debug.level=10
 debug.showcaller=false
 keys.ecc.curve.list=nistp256,nistp384,nistp521,sect163k1,nistk163,sect163r1,sect163r2,nistb163,sect193r1,sect193r2,sect233k1,nistk233,sect233r1,nistb233,sect239k1,sect283k1,nistk283,sect283r1,nistb283,sect409k1,nistk409,sect409r1,nistb409,sect571k1,nistk571,sect571r1,nistb571,secp160k1,secp160r1,secp160r2,secp192k1,secp192r1,nistp192,secp224k1,secp224r1,nistp224,secp256k1,secp256r1,secp384r1,secp521r1,prime192v1,prime192v2,prime192v3,prime239v1,prime239v2,prime239v3,c2pnb163v1,c2pnb163v2,c2pnb163v3,c2pnb176v1,c2tnb191v1,c2tnb191v2,c2tnb191v3,c2pnb208w1,c2tnb239v1,c2tnb239v2,c2tnb239v3,c2pnb272w1,c2pnb304w1,c2tnb359w1,c2pnb368w1,c2tnb431r1,secp112r1,secp112r2,secp128r1,secp128r2,sect113r1,sect113r2,sect131r1,sect131r2
 keys.ecc.curve.default=nistp521
diff -rupN pki-core-9.0.3.orig/base/common/src/com/netscape/certsrv/apps/CMS.java pki-core-9.0.3/base/common/src/com/netscape/certsrv/apps/CMS.java
--- pki-core-9.0.3.orig/base/common/src/com/netscape/certsrv/apps/CMS.java	2011-02-04 08:56:55.000000000 -0800
+++ pki-core-9.0.3/base/common/src/com/netscape/certsrv/apps/CMS.java	2014-06-19 21:10:16.000000000 -0700
@@ -78,9 +78,9 @@ import com.netscape.cmsutil.password.*;
  */
 public final class CMS {
 
-    public static final int DEBUG_OBNOXIOUS = 10;
+    public static final int DEBUG_OBNOXIOUS = 1;
     public static final int DEBUG_VERBOSE = 5;
-    public static final int DEBUG_INFORM = 1;
+    public static final int DEBUG_INFORM = 10;
 
     private static final String CONFIG_FILE = "CS.cfg";
     private static ICMSEngine _engine = null;
diff -rupN pki-core-9.0.3.orig/base/common/src/com/netscape/cmscore/util/Debug.java pki-core-9.0.3/base/common/src/com/netscape/cmscore/util/Debug.java
--- pki-core-9.0.3.orig/base/common/src/com/netscape/cmscore/util/Debug.java	2011-02-04 08:57:14.000000000 -0800
+++ pki-core-9.0.3/base/common/src/com/netscape/cmscore/util/Debug.java	2014-06-19 21:10:36.000000000 -0700
@@ -51,9 +51,9 @@ public class Debug
 	}
 
     public static final boolean ON = false;
-    public static final int OBNOXIOUS = 10;
+    public static final int OBNOXIOUS = 1;
     public static final int VERBOSE = 5;
-    public static final int INFORM = 1;
+    public static final int INFORM = 10;
 
     // the difference between this and 'ON' is that this is always
     // guaranteed to log to 'mOut', whereas other parts of the server
@@ -242,9 +242,9 @@ public class Debug
 
     /**
      * Set the current debugging level. You can use: <pre>
-     * OBNOXIOUS = 10
+     * OBNOXIOUS = 1
      * VERBOSE   = 5
-     * INFORM    = 1
+     * INFORM    = 10
      * </pre> Or another value
      */
 
